src/python/pants/backend/python/util_rules/faas.py

```diff
@@ -249,6 +267,49 @@ class PythonFaaSRuntimeField(StringField, ABC):
     def to_interpreter_version(self) -> None | tuple[int, int]:
         """Returns the Python version implied by the runtime, as (major, minor)."""
 
+    def to_platform_string(self) -> None | str:
+        # We hardcode the platform value to the appropriate one for each FaaS runtime.
+        # (Running the "hello world" cloud function in the example code will report the platform, and can be
+        # used to verify correctness of these platform strings.)
+        interpreter_version = self.to_interpreter_version()
+        if interpreter_version is None:
+            return None
+
+        py_major, py_minor = interpreter_version
+        platform_str = f"linux_x86_64-cp-{py_major}{py_minor}-cp{py_major}{py_minor}"
+        # set pymalloc ABI flag - this was removed in python 3.8 https://bugs.python.org/issue36707
+        if py_major <= 3 and py_minor < 8:
+            platform_str += "m"
+        return platform_str
+
+
+class PythonFaaSLayout(Enum):
+    # TODO: deprecate lambdex layout, since PEX can be used directly now
+    LAMBDEX = "lambdex"
+    ZIP = "zip"
+
+
+class PythonFaaSLayoutField(StringField):
+    alias = "layout"
+    valid_choices = PythonFaaSLayout
+    default = PythonFaaSLayout.LAMBDEX.value
+
+    help = help_text(
+        """
+        The layout used for the function artefact.
+
+        With the `lambdex` layout (default), the artefact is created as a Lambdex, which is a normal
+        PEX that's been adjusted to include a shim file for the handler. This requires dynamically
+        choosing dependencies on start-up.
+
+        With the `zip` layout, the artefact contains first and third party code at the top level,
```

Comment:
I think we can be more emphatic about the advantages of this, and say that it's the layout recommended by the cloud providers.